src/server/routes/units.ts

```diff
@@ -12,31 +12,22 @@ import checkPassword from '../services/authService';
 
 const router = express.Router();
 
-router.get('/units', (req: Request<unknown, unknown, unknown, {id: number | undefined}>, res: Response) => {
-  const { id } = req.query;
-  if (id === undefined) {
+router.get('/units', (req: Request<unknown, unknown, unknown, GetUnits>, res: Response) => {
+  const { query } = req;
+  if (query.id === undefined && query.parentId === undefined) {
     return getUnitsWithType(new Set([UnitType.BATTALION, UnitType.BRIGADE, UnitType.DIVISION]))
       .then((units: Unit[]) => res.status(200).send({ units }))
       .catch((error) => res.status(404).send({ errorMessage: error.message }));
+  } if (query.id !== undefined) {
+    return getUnit(query.id)
+      .then((unit: Unit) => res.status(200).send({ unit }))
+      .catch((error) => res.status(404).send({ errorMessage: error.message }));
+  } if (query.parentId !== undefined) {
+    return getChildUnits(query.parentId)
+      .then((units: Unit[]) => res.status(200).send({ units }))
+      .catch((error) => res.status(404).send({ errorMessage: error.message }));
   }
-  return getUnit(id)
-    .then((unit: Unit) => res.status(200).send({ unit }))
-    .catch((error) => res.status(404).send({ errorMessage: error.message }));
-});
-
-router.get('/units/:unitType', (req: Request<{unitType:string}, unknown, unknown, {parentId:number}>, res: Response) => {
-  const { unitType } = req.params;
-  const { parentId } = req.query;
-  if (parentId === undefined || unitType === undefined) {
-    return res.status(404).send({ errorMessage: 'Invalid request query.' });
-  }
-  const type = parseUnitType(unitType)!;
-  if (type === undefined) {
-    return res.status(404).send({ errorMessage: 'Invalid unitType.' });
-  }
-  return getChildUnits(parentId, type)
-    .then((units: Unit[]) => res.status(200).send({ units }))
-    .catch((error) => res.status(404).send({ errorMessage: error.message }));
+  return res.status(404).send({ errorMessage: 'Invalid Request.' });
```

Comment:
I thought we keep get /units/:unitType?parentId. Because this way the endpoint allows clients to get children companies of a division without fetching multiple time.

Reply:
Is that what we want? i thought we just need immediate child rather than grandchild? can @aaron-lam confirm? If yes, i will abandon this PR.

Reply:
@aaron-lam  said okay